### ISSUE TITLE
On receiving msg from room, user jid not-found error

### ISSUE THREAD

**Issue by ale-de** [Rainbow-CPaaS/node-red-contrib-ale-rainbow#7]
When receiving a message in a room, node-red debug shows this error:
"Rainbow : couldn't find user for jID : room_7002a54036f44ad38e99f32145b17103@muc.openrainbow.com/b9481e713a194e10844ea1d18445bb9a@openrainbow.com"
This also prevents the messages being sent on the msg.payload.

**Comment by ArfyFR** [comment]
Hi,

Could you please in the debug traces
`node-red -v
`
first give me the versions informations ?

Like:
```
2018-3-28 10:25:07 [1522225507010] - debug: CORE - (constructor) ------- SDK INFORMATION -------
2018-3-28 10:25:07 [1522225507016] - info: CORE -  (constructor) SDK version: 1.37.0
2018-3-28 10:25:07 [1522225507029] - info: CORE -  (constructor) Node version: v8.9.4
2018-3-28 10:25:07 [1522225507041] - info: CORE -  (constructor) http_parser version: 2.7.0
2018-3-28 10:25:07 [1522225507044] - info: CORE -  (constructor) node version: 8.9.4
2018-3-28 10:25:07 [1522225507050] - info: CORE -  (constructor) v8 version: 6.1.534.50
2018-3-28 10:25:07 [1522225507060] - info: CORE -  (constructor) uv version: 1.15.0
2018-3-28 10:25:07 [1522225507076] - info: CORE -  (constructor) zlib version: 1.2.11
2018-3-28 10:25:07 [1522225507081] - info: CORE -  (constructor) ares version: 1.10.1-DEV
2018-3-28 10:25:07 [1522225507115] - info: CORE -  (constructor) modules version: 57
2018-3-28 10:25:07 [1522225507140] - info: CORE -  (constructor) nghttp2 version: 1.25.0
2018-3-28 10:25:07 [1522225507161] - info: CORE -  (constructor) openssl version: 1.0.2n
2018-3-28 10:25:07 [1522225507170] - info: CORE -  (constructor) icu version: 59.1
2018-3-28 10:25:07 [1522225507174] - info: CORE -  (constructor) unicode version: 9.0
2018-3-28 10:25:07 [1522225507181] - info: CORE -  (constructor) cldr version: 31.0.1
2018-3-28 10:25:07 [1522225507197] - info: CORE -  (constructor) tz version: 2017b
```

My own "test bot" in prod is working.

Best regards,
Pascal

**Comment by ale-de** [comment]
Pascal,


Because it stopped my development, I already patched rainbow.js from 'node-red-contrib-ale-rainbow' to work around this issue, see below in highlighted yellow.

But I guess you need to check first whether msg.type is a groupchat or not. I've also added the node-red log file as you asked.


Regards


Dirk

  function getMessage(config) {
    RED.nodes.createNode(this,config);
    this.filter = config.filter;
    this.filterContact = config.filterContact;
    this.filterCompany = config.filterCompany;
    this.server = RED.nodes.getNode(config.server);
    serverctx = this.server.context();

    var cfgTimer = null;
    var node = this;
    node.log("Rainbow : getMessage node initialized :" + this.id)

    var rainbow_onmessagereceivedGetMessage = function rainbow_onmessagereceivedGetMessage(message) {
      let filterOK = true;
      node.log("Rainbow : onMessageReceived")
      node.log("Rainbow : Message :"+JSON.stringify(message));

      // RegExp filter ?
      if ((node.filter != '') && (node.filter != undefined)) {
        var regexp = new RegExp(node.filter, 'img');
        node.log("Rainbow : Apply filter :"+node.filter);
        var res = JSON.stringify(message.content).match(regexp);
        node.log("Rainbow : Filter result:"+JSON.stringify(res))
        if (res == null) {
          filterOK = false;
          node.log("Rainbow : RegExp filter blocked !");
          return;
        }
      }

      // Get contact information from server
      rainbowSDK[config.server]
//        ._core
//        ._contacts
      .contacts
      .getContactByJid(message.fromJid)
      .then((contact) => {
        if (contact) {

          // Contact filter ?
          if ( (node.filterContact != '') && (node.filterContact != undefined) ) {
            filterOK = false;
            if ( (node.filterContact === contact.loginEmail) || (node.filterContact === contact.jid_im) ) {
              filterOK = true;
              node.log("Rainbow : Contact filter OK !");
            } else {
              node.log("Rainbow : Contact filter blocked !");
              return;
            }
          }

          // Company filter ?
          if ( (node.filterCompany != '') && (node.filterCompany != undefined) ) {
            filterOK = false;
            if (node.filterCompany === contact.companyId) {
              filterOK = true;
              node.log("Rainbow : Company id filter OK !");
            } else {
              node.log("Rainbow : Company id filter blocked !");
              return;
            }
          }

          node.log("Rainbow : sending message.");

          var msg = { payload: message, contact: contact };
          node.send(msg);

        } else {
          node.warn("Rainbow : couldn't find user for jID : " + message.fromJid);
            var msg = { payload: message, contact: {} };
            node.send(msg);
        }

      });

    };


## Node-red log file

Welcome to Node-RED

===================


28 Mar 10:42:28 - [info] Node-RED version: v0.18.4

28 Mar 10:42:28 - [info] Node.js  version: v8.9.3

28 Mar 10:42:28 - [info] Linux 4.9.65-1-hardened x64 LE

28 Mar 10:42:29 - [info] Loading palette nodes

28 Mar 10:42:31 - [info] Dashboard version 2.8.2 started at /ui

28 Mar 10:42:33 - [warn] ------------------------------------------------------

28 Mar 10:42:33 - [warn] [node-red/rpi-gpio] Info : Ignoring Raspberry Pi specific node

28 Mar 10:42:33 - [warn] ------------------------------------------------------

28 Mar 10:42:33 - [info] Settings file  : /root/.node-red/settings.js

28 Mar 10:42:33 - [info] User directory : /root/.node-red

28 Mar 10:42:33 - [warn] Projects disabled : set editorTheme.projects.enabled=true to enable

28 Mar 10:42:33 - [info] Flows file     : /root/.node-red/flows_aleserver.json

28 Mar 10:42:33 - [info] Server now running at http://127.0.0.1:1880/

28 Mar 10:42:33 - [info] Starting flows

28 Mar 10:42:33 - [info] [Login:Bot1] ******************** IN login

28 Mar 10:42:33 - [info] [Login:Bot1] Rainbow : login node initialized :a16c21ac.f73a8

28 Mar 10:42:33 - [info] [Login:Bot1] Rainbow : login node launch require **********

28 Mar 10:42:34 - [info] [Login:Bot1] config : {"id":"a16c21ac.f73a8","type":"Login","z":"","name":"Bot1","autoLogin":true,"proxyHost":"","proxyPort":"","proxyProto":"","ackIM":false,"_users":["69903465.c1aebc","43162fc6.ded7e"]}

28 Mar 10:42:34 - [info] [Login:Bot1] ENV : http_proxy=undefined

28 Mar 10:42:34 - [info] [Login:Bot1] ENV : https_proxy=undefined

2018-3-28 08:42:34 [1522226554110] - info: LOGS - ------------------------------------------------

2018-3-28 08:42:34 [1522226554125] - info: LOGS - Welcome to the ALE Rainbow SDK for Node.JS

2018-3-28 08:42:34 [1522226554126] - info: LOGS - Where Everything connects

2018-3-28 08:42:34 [1522226554127] - info: LOGS - Support: Send message to Emily using #support #api

2018-3-28 08:42:34 [1522226554127] - info: LOGS - ------------------------------------------------

2018-3-28 08:42:34 [1522226554128] - info: LOGS - (constructor) No file logs enabled

2018-3-28 08:42:34 [1522226554129] - debug: CORE - (constructor) _entering_

2018-3-28 08:42:34 [1522226554129] - debug: CORE - (constructor) ------- SDK INFORMATION -------

2018-3-28 08:42:34 [1522226554130] - info: CORE -  (constructor) SDK version: 1.38.0

2018-3-28 08:42:34 [1522226554130] - info: CORE -  (constructor) Node version: v8.9.3

2018-3-28 08:42:34 [1522226554131] - info: CORE -  (constructor) http_parser version: 2.7.1

2018-3-28 08:42:34 [1522226554131] - info: CORE -  (constructor) node version: 8.9.3

2018-3-28 08:42:34 [1522226554132] - info: CORE -  (constructor) v8 version: 6.1.534.48

2018-3-28 08:42:34 [1522226554132] - info: CORE -  (constructor) uv version: 1.17.0

2018-3-28 08:42:34 [1522226554132] - info: CORE -  (constructor) zlib version: 1.2.11

2018-3-28 08:42:34 [1522226554133] - info: CORE -  (constructor) ares version: 1.13.0

2018-3-28 08:42:34 [1522226554133] - info: CORE -  (constructor) modules version: 57

2018-3-28 08:42:34 [1522226554134] - info: CORE -  (constructor) nghttp2 version: 1.25.0

2018-3-28 08:42:34 [1522226554134] - info: CORE -  (constructor) openssl version: 1.0.2m

2018-3-28 08:42:34 [1522226554135] - info: CORE -  (constructor) icu version: 59.1

2018-3-28 08:42:34 [1522226554135] - info: CORE -  (constructor) unicode version: 9.0

2018-3-28 08:42:34 [1522226554136] - info: CORE -  (constructor) cldr version: 31.0.1

2018-3-28 08:42:34 [1522226554136] - info: CORE -  (constructor) tz version: 2017b

2018-3-28 08:42:34 [1522226554136] - debug: CORE - (constructor) ------- SDK INFORMATION -------

2018-3-28 08:42:34 [1522226554138] - error: OPTIONS - (constructor) application' property is not defined. Can't sign-in. Check the documentation to configure it

2018-3-28 08:42:34 [1522226554139] - debug: OPTIONS - (constructor) Use REST services on Rainbow Official platform

2018-3-28 08:42:34 [1522226554140] - debug: OPTIONS - (constructor) Use XMPP services on Rainbow Official platform

2018-3-28 08:42:34 [1522226554141] - warn: OPTIONS - (constructor) 'appID' property is not defined. No application ID will be used

2018-3-28 08:42:34 [1522226554142] - warn: OPTIONS - (constructor) 'appSecret' property is not defined. No application Secret will be used

2018-3-28 08:42:34 [1522226554144] - info: CORE - (constructor) used in XMPP mode

2018-3-28 08:42:34 [1522226554155] - debug: CORE - (constructor) _exiting_

28 Mar 10:42:34 - [info] [Login:Bot1] RainbowSDK instance a16c21ac.f73a8 allocated

28 Mar 10:42:34 - [info] [Login:Bot1] Rainbow : login register for event 'rainbow_onready'

28 Mar 10:42:34 - [info] [Login:Bot1] Rainbow : login register for event 'rainbow_onstopped'

28 Mar 10:42:34 - [info] [Login:Bot1] Rainbow : login register for event 'rainbow_onconnectionerror'

28 Mar 10:42:34 - [info] [Login:Bot1] Rainbow : login register for event 'rainbow_onerror'

28 Mar 10:42:34 - [info] [Login:Bot1] RainbowSDK instance for autostart a16c21ac.f73a8

28 Mar 10:42:35 - [info] [Notified_IM:Bot1] Rainbow : getMessage node initialized :69903465.c1aebc

28 Mar 10:42:35 - [info] [Notified_IM:Bot1] Rainbow SDK (a16c21ac.f73a8) Registering rainbow_onmessagereceived

28 Mar 10:42:35 - [info] [Notified_IM:Bot1] Rainbow : getMessage register for event 'rainbow_onmessagereceived'

28 Mar 10:42:35 - [info] [CnxState:CnxState Bot1] Rainbow : getCnxState node initialized :43162fc6.ded7e

28 Mar 10:42:35 - [info] [CnxState:CnxState Bot1] Rainbow : getCnxState register for event 'rainbow_onconnected'

28 Mar 10:42:35 - [info] [CnxState:CnxState Bot1] Rainbow : getCnxState register for event 'rainbow_onconnectionerror'

28 Mar 10:42:35 - [info] [CnxState:CnxState Bot1] Rainbow : getCnxState register for event 'rainbow_onerror'

28 Mar 10:42:35 - [info] [CnxState:CnxState Bot1] Rainbow : getCnxState register for event 'rainbow_onready'

28 Mar 10:42:35 - [error] [ov2500-wlanclientlist:c8471372.b4802] Entering wlanclientlist

28 Mar 10:42:35 - [error] [ov2500-about:7c46aa1a.d5ad04] Entering about

28 Mar 10:42:35 - [info] Started flows




Dirk Evrard
Director Presales Verticals - EUNO
+32 475 611669  UC: +32 3 2910826
dirk.evrard@al-enterprise.com
[1497536993982_banner.png]
________________________________
Van: Pascal Borschneck <notifications@github.com>
Verzonden: woensdag 28 maart 2018 10:29:57
Aan: Rainbow-CPaaS/node-red-contrib-ale-rainbow
CC: Evrard Dirk; Author
Onderwerp: EXT: Re: [Rainbow-CPaaS/node-red-contrib-ale-rainbow] On receiving msg from room, user jid not-found error (#7)


Hi,

Could you please in the debug traces
node-red -v
first give me the versions informations ?

Like:

2018-3-28 10:25:07 [1522225507010] - debug: CORE - (constructor) ------- SDK INFORMATION -------
2018-3-28 10:25:07 [1522225507016] - info: CORE -  (constructor) SDK version: 1.37.0
2018-3-28 10:25:07 [1522225507029] - info: CORE -  (constructor) Node version: v8.9.4
2018-3-28 10:25:07 [1522225507041] - info: CORE -  (constructor) http_parser version: 2.7.0
2018-3-28 10:25:07 [1522225507044] - info: CORE -  (constructor) node version: 8.9.4
2018-3-28 10:25:07 [1522225507050] - info: CORE -  (constructor) v8 version: 6.1.534.50
2018-3-28 10:25:07 [1522225507060] - info: CORE -  (constructor) uv version: 1.15.0
2018-3-28 10:25:07 [1522225507076] - info: CORE -  (constructor) zlib version: 1.2.11
2018-3-28 10:25:07 [1522225507081] - info: CORE -  (constructor) ares version: 1.10.1-DEV
2018-3-28 10:25:07 [1522225507115] - info: CORE -  (constructor) modules version: 57
2018-3-28 10:25:07 [1522225507140] - info: CORE -  (constructor) nghttp2 version: 1.25.0
2018-3-28 10:25:07 [1522225507161] - info: CORE -  (constructor) openssl version: 1.0.2n
2018-3-28 10:25:07 [1522225507170] - info: CORE -  (constructor) icu version: 59.1
2018-3-28 10:25:07 [1522225507174] - info: CORE -  (constructor) unicode version: 9.0
2018-3-28 10:25:07 [1522225507181] - info: CORE -  (constructor) cldr version: 31.0.1
2018-3-28 10:25:07 [1522225507197] - info: CORE -  (constructor) tz version: 2017b


My own "test bot" in prod is working.

Best regards,
Pascal

**Comment by ArfyFR** [comment]
Hi,

Ok, got it thanks, "_my bad_", I didn't play/test with a muc (_multi user chat_) since I added contact information ;)
A new version will be released asap.

**Comment by ale-de** [comment]
Great!! Can you also add the bubble (room), next to the contact in the result msg? This makes it easier to make decisions based on eg bubble name.


tx


Dirk


Dirk Evrard
Director Presales Verticals - EUNO
+32 475 611669  UC: +32 3 2910826
dirk.evrard@al-enterprise.com
[1497536993982_banner.png]
________________________________
Van: Pascal Borschneck <notifications@github.com>
Verzonden: woensdag 28 maart 2018 15:35:40
Aan: Rainbow-CPaaS/node-red-contrib-ale-rainbow
CC: Evrard Dirk; Author
Onderwerp: EXT: Re: [Rainbow-CPaaS/node-red-contrib-ale-rainbow] On receiving msg from room, user jid not-found error (#7)


Hi,

Ok, got it thanks, "my bad", I didn't play/test with a muc (multi user chat) since I added contact information ;)
A new version will be released asap.

**Comment by ArfyFR** [comment]
Hi,

I added in the doc, information on type : "chat" or "groupchat" and also 2 checkboxes to filter them or not.

And in case of groupchat,

- the sender jid is already in fromBubbleUserJid.
- I am adding the bubble also to the payload as asked.

**Comment by ArfyFR** [comment]
Corrected in v1.2.8
